File: iCyption2022/CftTest/ikut/solution.cpp
#include <iostream>

using namespace std;

struct peserta {
    string nama;
    int jmlSoal;
};

int main() {
    int a, b;
    cin >> a;
    peserta p[100];
    for (int i=0; i<a; ++i) {
        cin >> p[i].nama >> p[i].jmlSoal;
    }
    cin >> b;

    int min = 101;
    string guru;
    for (int i=0; i<a; ++i) {
        if (p[i].jmlSoal<min && p[i].jmlSoal>=b) {
            guru = p[i].nama;
            min = p[i].jmlSoal;
        }
    }

    if (min==101) {
        cout << "Tidak bisa berguru";
    } else {
        cout << guru;
    }

    return 0;
}

File: iCyption2022/CftTest/ikut/solution.py
a = int(input())
peserta = []
for i in range(a):
    nama, jmlSoal = input().split()
    jmlSoal = int(jmlSoal)
    peserta.append({"nama":nama, "jmlSoal":jmlSoal})
b = int(input())

min = 101
for p in peserta:
    if p["jmlSoal"]<min and p["jmlSoal"]>=b:
        guru = p["nama"]
        min = p["jmlSoal"]

if min==101:
    print("Tidak bisa berguru", end="")
else:
    print(guru, end="")
